### PR TITLE
fix: useAttachments use the correct functions, for adding and removing

### PR DESCRIPTION
**Pull request by ragi96**
https://app.clickup.com/t/8693nbz40

### CODE REVIEW

File: .changeset/afraid-trees-mate.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@frontify/guideline-blocks-settings": patch
+---
+
+fix(Attachments): use the according function for removing and adding asset
```
Comment (SamCreasey) [inline .changeset/afraid-trees-mate.md:5]:
```suggestion
fix(Attachments): use the appropriate function for removing and adding asset
```
Maybe?

File: packages/guideline-blocks-settings/src/hooks/useAttachments.tsx [hunk 2]
```diff
@@ -6,23 +6,18 @@ import { type ReactNode, createContext, useContext } from 'react';
...
+        );
     };
 
     const onAttachmentDelete = async (assetToDelete: Asset) => {
-        const newAssetIds = attachments
-            .filter((attachment) => attachment.id !== assetToDelete.id)
-            .map((attachment) => attachment.id);
-
-        await updateAssetIdsFromKey(assetId, newAssetIds);
+        await deleteAssetIdsFromKey(assetId, [assetToDelete.id]);
```
Comment (SamCreasey) [inline packages/guideline-blocks-settings/src/hooks/useAttachments.tsx:20]: Looking at the function names should we rename the argument `assetId` in the useAttachments hook? it seems odd to call deleteAssetIdsFromKey with the first argument being `assetId`, which is actually the `key` in this case

**Review by SamCreasey — APPROVED** [review]
Functionally it works in the audio block, just left two minor comments that are just suggestions here

File: .changeset/afraid-trees-mate.md (hunk 1 above)
Comment (ragi96) [reply .changeset/afraid-trees-mate.md:5]: Seems better :)

File: packages/guideline-blocks-settings/src/hooks/useAttachments.tsx (hunk 2 above)
Comment (ragi96) [reply packages/guideline-blocks-settings/src/hooks/useAttachments.tsx]: changed to `attachmentKey`

**Pull request merged**